chainerx_cc/chainerx/cuda/cudnn.h

```diff
@@ -10,21 +12,40 @@
 #include "chainerx/macro.h"
 #include "chainerx/stack_vector.h"
 
+#define CHAINERX_CUDA_CUDNN_CALL_WITH_HANDLE(handle, func, args...)                                                                    \
+    {                                                                                                                                  \
+        static_assert(std::is_same<std::remove_reference_t<decltype(handle)>, chainerx::cuda::cuda_internal::CudnnHandle>::value, ""); \
+        handle.Call(func, #func, args);                                                                                                \
+    }
+
+#define CHAINERX_CUDA_CUDNN_CALL(func, args...) chainerx::cuda::cudnn_detail::CheckCudnnError(#func, func(args))
+
 namespace chainerx {
 namespace cuda {
 
 class CudnnError : public ChainerxError {
 public:
     using ChainerxError::ChainerxError;
+};
 
-    explicit CudnnError(cudnnStatus_t status);
+class CudnnStatusError : public CudnnError {
+public:
+    CudnnStatusError(const char* func_name, cudnnStatus_t status);
     cudnnStatus_t error() const noexcept { return status_; }
 
 private:
     cudnnStatus_t status_{};
 };
 
-void CheckCudnnError(cudnnStatus_t status);
+namespace cudnn_detail {
+
+inline void CheckCudnnError(const char* func_name, cudnnStatus_t status) {
```

Comment:
A detail but how about passing `func_name` as the later argument as it's less significant (in most cases never used).

Reply:
What do you mean by never used? It's mandatory.

Reply:
It's only used when the condition evaluates to true, that's what I meant...

Reply:
Oh does that matter? `func_name` is more "stable" parameter, so I think it's natural to come first.

Reply:
I thought so. Or at least, I'm not sure what you mean with "stable", but the intent of this functions is to check the status, doesn't `status` have a higher importance? 🤔